### PR TITLE
Add support for Redis cluster mode

### PR DESCRIPTION
**Pull request by jaspervanveghel**
👋 Hi folks. We were previously maintaining [tc-redis-cluster](https://github.com/dffrntmedia/tc-redis-cluster) and are in the process of upgrading to Thumbor 7. This PR adds Redis cluster-mode support; which is now also officially supported by [redis-py](https://github.com/redis/redis-py), making this a whole lot easier!

### CODE REVIEW

File: tc_redis/base_storage.py [hunk 1]
```diff
@@ -6,14 +6,18 @@
 # Copyright (c) 2022 Raphael Rossi <raphael.vieira.rossi@gmail.com>
 
 from redis import Redis, Sentinel
+from redis.cluster import RedisCluster
+from redis.cluster import ClusterNode
```
Comment (RaphaelVRossi) [inline tc_redis/base_storage.py]: You can change these two lines 
```python
from redis.cluster import RedisCluster
from redis.cluster import ClusterNode
```
for this
```python
from redis.cluster import RedisCluster, ClusterNode
```

**Review by RaphaelVRossi — CHANGES_REQUESTED** [review]
Hey! 

Thank you for contribute!

You can use this PR #9 as tests examples.

After this, for me, it's all good!

**Comment by mbarouski** [comment]
Hi @RaphaelVRossi 👋🏻  Tests are added in [this PR](https://github.com/jaspervanveghel/redis/pull/1/). Waiting for approval from @jaspervanveghel...

**Comment by mbarouski** [comment]
Ping @RaphaelVRossi 🏓

File: tests/test_redis_cluster_result_storage.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    async def test_should_raises_attribute_error(self):
+        with self.assertRaises(AttributeError) as error:
+            RedisStorage(self.ctx)
+
+        expect(str(error.exception)).to_equal(
+            "Unknown value for REDIS_RESULT_STORAGE_MODE test. See README for more information."
+        )
+
+
+class ConnectToSentinelDatabase(RedisDBContext):
```
Comment (RaphaelVRossi) [inline tests/test_redis_cluster_result_storage.py]: Hey @mbarouski, did you forget this sentinel test inside your file?!

**Review by RaphaelVRossi — CHANGES_REQUESTED** [review]

File: tests/test_redis_cluster_result_storage.py (hunk 2 above)
Comment (mbarouski) [reply tests/test_redis_cluster_result_storage.py]: Oops... Removed ✔️ 
Waiting for approval from @jaspervanveghel...

Comment (RaphaelVRossi) [reply tests/test_redis_cluster_result_storage.py]: Thank you! Now LGTM! 🥳

**Comment by mbarouski** [comment]
@RaphaelVRossi should we wait for one more approval from somebody else to merge this?

**Comment by RaphaelVRossi** [comment]
> @RaphaelVRossi should we wait for one more approval from somebody else to merge this?

@devppjr Could you please review?!

**Review by devppjr — APPROVED** [review]
LGTM

**Comment by devppjr** [comment]
2.3.0 is live!

https://pypi.org/project/tc-redis/2.3.0/

**Pull request merged**